src/sql/util.py

```diff
@@ -360,3 +363,22 @@ def show_deprecation_warning():
         "raise an exception in the next major release so please remove it.",
         FutureWarning,
     )
+
+
+def _check_table_exists(table, schema=None):
```

Comment:
I'd give this a more informative name

```suggestion
def is_saved_snippet_or_table_exists(table, schema=None):
```

Reply:
Assuming this is the function name `is_saved_snippet_or_table_exists`, I would expect it to return a boolean value, whether there is a snippet, table, or schema with such a name. Both `is_table_exists` and `is_saved_snippet` return a boolean, so it's a little bit confusing that we are returning `with_` which might be `None` or a list with the table name.

Another point is that `is_table_exists` throws an error by default if the table doesn't exist with suggestions based on current tables and snippets.

<img width="751" alt="image" src="https://github.com/ploomber/jupysql/assets/103026689/6e3f9d9e-491e-4cd9-94c3-25c3b2f849e0">


I'm not sure what's the best approach here. Maybe a different name `check_saved_snippet_or_table_exists` and it will return a type, snippet/table? or we can return 2 values, `is_exist` and `is_snippet` (which also might confuse since the function throws an error by default and `is_exist` is irrelevant for most of the cases unless we use the `ignore_error` flag).

@edublancas thoughts?

Reply:
> Assuming this is the function name `is_saved_snippet_or_table_exists`, I would expect it to return a boolean value, whether there is a snippet, table, or schema with such a name. Both `is_table_exists` and `is_saved_snippet` return a boolean, so it's a little bit confusing that we are returning `with_` which might be `None` or a list with the table name.
> 
> Another point is that `is_table_exists` throws an error by default if the table doesn't exist with suggestions based on current tables and snippets.
> 
> <img alt="image" width="751" src="https://user-images.githubusercontent.com/103026689/252914663-6e3f9d9e-491e-4cd9-94c3-25c3b2f849e0.png">
> I'm not sure what's the best approach here. Maybe a different name `check_saved_snippet_or_table_exists` and it will return a type, snippet/table? or we can return 2 values, `is_exist` and `is_snippet` (which also might confuse since the function throws an error by default and `is_exist` is irrelevant for most of the cases unless we use the `ignore_error` flag).
> 
> @edublancas thoughts?

I have renamed the function for now